File: ros_projects/src/publisher_subscriber/py/message_subscriber.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import String

def callback_message(message):
    #get id and print message
    rospy.loginfo(rospy.get_caller_id() + "Recieved Message: %s", message.data)
    

if __name__ == '__main__':
    # Initialize node as subscriber
    rospy.init_node('subscriber', anonymous=True)
    # Define topic which will be listened
    rospy.Subscriber("pub_message", String, callback_message)
    # Infinite loop
    rospy.spin()

File: src/ros_projects/src/publisher_subscriber/cpp/message_subscriber.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

void callback_message(const std_msgs::String::ConstPtr&);

int main(int argc, char** argv)
{
	// Initializing ros
	ros::init(argc, argv, "sub_node");

	// Defining node to include subscriber
	ros::NodeHandle sub_node;
	
	// Defning subscriber to subscribe pub_message topic
	ros::Subscriber _sub = sub_node.subscribe("pub_message", 1000, callback_message);
	
	// Infinite loop
	ros::spin();

	return 0;
}

void callback_message(const std_msgs::String::ConstPtr& message)
{
	ROS_INFO("Message Recieved: %s\n", message->data.c_str());
}
